data_structures/trie_using_hashmap.cpp

```diff
@@ -0,0 +1,330 @@
+/**
+ * @file
+ * @author [Venkata Bharath](https://github.com/bharath000)
+ * @brief Implementation of [Trie](https://en.wikipedia.org/wiki/Trie) data
+ * structure using HashMap for different characters and method for predicting
+ * words based on prefix.
+ * @details The Trie data structure is implemented using unordered map to use
+ * memory optimally, predict_words method is developed to recommend words based
+ * on a given prefix along with other methods insert, delete, search, startwith
+ * in trie.
+ * @see trie_modern.cpp for difference
+ */
+
+#include <cassert>        /// for assert
+#include <iostream>       /// for IO operations
+#include <memory>         /// for std::shared_ptr
+#include <stack>          /// for std::stack
+#include <unordered_map>  /// for std::unordered_map
+#include <vector>         /// for std::vector
+
+/**
+ * @namespace data_structures
+ */
+
+namespace data_structures {
+/**
+ * Class documentation
+ */
+class Trie {
+ private:
+    struct Node {
+        /** @brief trie nodes as key value pairs, value is a shared_pointer of
```

Comment:
Is this the documentation of the `Node` struct?

Reply:
updated documentation to Node struct